plugins/network-elements/nuage-vsp/src/com/cloud/network/resource/NuageVspResourceConfiguration.java

```diff
@@ -0,0 +1,310 @@
+
+    private String _name;
+    private String _guid;
+    private String _zoneId;
+    private String _hostName;
+    private String _cmsUser;
+    private String _cmsUserPassword;
+    private String _port;
+    private String _apiVersion;
+    private String _apiRelativePath;
+    private String _retryCount;
+    private String _retryInterval;
+    private String _nuageVspCmsId;
+
+    public String name() {
+        return _name;
+    }
+
+    public String guid() {
+        return this._guid;
+    }
+
+    public NuageVspResourceConfiguration guid(String guid) {
+        this._guid = guid;
+        return this;
+    }
+
+    public String zoneId() {
+        return this._zoneId;
+    }
+
+    public NuageVspResourceConfiguration zoneId(String zoneId) {
+        this._zoneId = zoneId;
+        return this;
+    }
+
+    public String hostName() {
+        return this._hostName;
+    }
+
+    public NuageVspResourceConfiguration hostName(String hostName) {
+        this._hostName = hostName;
+        this._name = "Nuage VSD - " + _hostName;
+        return this;
+    }
+
+    public String cmsUser() {
+        return this._cmsUser;
+    }
+
+    public NuageVspResourceConfiguration cmsUser(String cmsUser) {
+        this._cmsUser = cmsUser;
+        return this;
+    }
+
+    public String cmsUserPassword() {
+        return this._cmsUserPassword;
+    }
+
+    public NuageVspResourceConfiguration cmsUserPassword(String cmsUserPassword) {
+        this._cmsUserPassword = cmsUserPassword;
+        return this;
+    }
+
+    public String port() {
+        return this._port;
+    }
+
+    public NuageVspResourceConfiguration port(String port) {
+        this._port = port;
+        return this;
+    }
+
+    public String apiVersion() {
+        return this._apiVersion;
+    }
+
+    public NuageVspResourceConfiguration apiVersion(String apiVersion) {
+        this._apiVersion = apiVersion;
+        return this;
+    }
+
+    public String apiRelativePath() {
+        return this._apiRelativePath;
+    }
+
+    public NuageVspResourceConfiguration apiRelativePath(String apiRelativePath) {
+        this._apiRelativePath = apiRelativePath;
+        return this;
+    }
+
+    public String retryCount() {
+        return this._retryCount;
+    }
+
+    public NuageVspResourceConfiguration retryCount(String retryCount) {
+        this._retryCount = retryCount;
+        return this;
+    }
+
+    public String retryInterval() {
+        return this._retryInterval;
+    }
+
+    public NuageVspResourceConfiguration retryInterval(String retryInterval) {
+        this._retryInterval = retryInterval;
+        return this;
+    }
+
+    public String nuageVspCmsId() {
+        return this._nuageVspCmsId;
+    }
+
+    public NuageVspResourceConfiguration nuageVspCmsId(String nuageVspCmsId) {
+        this._nuageVspCmsId = nuageVspCmsId;
+        return this;
+    }
+
+    public String getRootPath(){
```

Comment:
NuageVspResourceConfiguration is used as a field in UpdateNuageVspDeviceCommand.
As this might be (de)serialized to Json is the resource is on a different management server in the cluster, and URI doesn't have a Gson TypeAdapter.